exporter/sapmexporter/exporter.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package sapmexporter exports trace data using Splunk's SAPM protocol.
+package sapmexporter
+
+import (
+	"context"
+	"errors"
+
+	"github.com/open-telemetry/opentelemetry-collector/consumer/consumerdata"
+	"github.com/open-telemetry/opentelemetry-collector/consumer/consumererror"
+	"github.com/open-telemetry/opentelemetry-collector/exporter"
+	"github.com/open-telemetry/opentelemetry-collector/exporter/exporterhelper"
+	"github.com/open-telemetry/opentelemetry-collector/translator/trace/jaeger"
+	sapmclient "github.com/signalfx/sapm-proto/client"
+	"go.uber.org/zap"
+)
+
+// sapmExporter is a wrapper struct of SAPM exporter
+type sapmExporter struct {
+	client *sapmclient.Client
+	logger *zap.Logger
+}
+
+func (*sapmExporter) Name() string {
+	return "sapm"
+}
+
+func (se *sapmExporter) Shutdown() error {
+	se.client.Stop()
+	return nil
+}
+
+func newSAPMTraceExporter(cfg *Config, logger *zap.Logger) (exporter.TraceExporter, error) {
+	if cfg.Endpoint == "" {
+		return nil, errors.New("`endpoint` not specified")
+	}
+
+	opts := []sapmclient.Option{
+		sapmclient.WithEndpoint(cfg.Endpoint),
+	}
+	if cfg.NumWorkers > 0 {
+		opts = append(opts, sapmclient.WithWorkers(cfg.NumWorkers))
+	}
+
+	if cfg.MaxConnections > 0 {
+		opts = append(opts, sapmclient.WithMaxConnections(cfg.MaxConnections))
+	}
+
+	if cfg.MaxRetries != nil {
+		opts = append(opts, sapmclient.WithMaxRetries(*cfg.MaxRetries))
+	}
+
+	if cfg.AuthToken != "" {
+		opts = append(opts, sapmclient.WithAuthToken(cfg.AuthToken))
+	}
+
+	client, err := sapmclient.New(opts...)
+	if err != nil {
+		return nil, err
+	}
+	se := sapmExporter{
+		client: client,
+		logger: logger,
+	}
+	return exporterhelper.NewTraceExporter(
+		cfg,
+		se.pushTraceData,
+		exporterhelper.WithTracing(true),
+		exporterhelper.WithMetrics(true),
+		exporterhelper.WithShutdown(se.Shutdown))
+}
+
+func (se *sapmExporter) pushTraceData(ctx context.Context, td consumerdata.TraceData) (int, error) {
+	jBatch, err := jaeger.OCProtoToJaegerProto(td)
+	if err != nil {
+		return 0, consumererror.Permanent(err)
+	}
+	err = se.client.Export(ctx, jBatch)
```

Comment:
@owais as you know in general is better to avoid batching and retries on the exporter, not a deal breaker but not recommended in general. Any plans to make it a simple sender down the road?